Cyrano/src/tools/rag-query.ts

```
@@ -161,6 +186,48 @@ All retrieved information includes source attribution for transparency and verif
+          // Process documents in parallel with Promise.allSettled to handle individual failures
+          const results = await Promise.allSettled(
+            documents.map(doc => 
+              this.ragService.ingestDocument({
+                id: doc.id,
+                text: doc.text,
+                type: doc.type,
+                source: doc.source,
+                sourceType: doc.sourceType,
+              })
+            )
+          );
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)